contrib/nydus-snapshotter/pkg/auth/keychain.go

```diff
@@ -61,10 +61,20 @@ func (kc PassKeyChain) TokenBase() bool {
 
 // FromLabels find image pull username and secret from snapshot labels
 // if username and secret is empty, we treat it as empty string
-func FromLabels(labels map[string]string) PassKeyChain {
-	return PassKeyChain{
-		Username: labels[label.ImagePullUsername],
-		Password: labels[label.ImagePullSecret],
+func FromLabels(labels map[string]string) *PassKeyChain {
+	u, found := labels[label.ImagePullUsername]
+	if !found {
```

Comment:
should check for empty username/password as well.

Reply:
Sure. It now also checks if the values are void string.